packages/video_player/lib/video_player.dart

```diff
@@ -132,22 +132,27 @@ class VideoPlayerValue {
 /// After [dispose] all further calls are ignored.
 class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   int _textureId;
-  final String uri;
   Timer timer;
   bool isDisposed = false;
   Completer<Null> _creatingCompleter;
   StreamSubscription<dynamic> _eventSubscription;
   _VideoAppLifeCycleObserver _lifeCycleObserver;
+  dynamic _createMethodArg;
```

Comment:
It is a bit sad that we need to carry this map around for the lifetime of the controller, given that it is only used during initialization. What about having `final String dataSource` and `final bool isNetwork` fields instead? Those may be useful for other purposes. Error reporting on playback failure should be very different for a network datasource compared to an asset one.